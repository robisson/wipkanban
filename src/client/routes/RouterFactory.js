import React from 'react';
import {StaticRouter} from 'react-router';
import {BrowserRouter as Router} from 'react-router-dom'
import Routes from './Routes';

const RouterFactory = ({server}) => {

    if (server) {
        return (
            <StaticRouter>
                <Routes/>
            </StaticRouter>
        )
    } else {
        return (
            <Router>
                <Routes/>
            </Router>
        )
    }

};

export default RouterFactory;