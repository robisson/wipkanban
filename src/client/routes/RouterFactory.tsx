// @flow
import * as React from "react";
import { StaticRouter } from "react-router";
import { BrowserRouter } from "react-router-dom";
import Routes from "./Routes";

interface IRouterFactoryProps {
  server?: boolean;
  location?: string;
}

const RouterFactory = ({ server, location }: IRouterFactoryProps) => {
  if (server) {
    return (
      <StaticRouter location={location} context={{}}>
        <Routes />
      </StaticRouter>
    );
  }

  return (
    <BrowserRouter>
      <Routes />
    </BrowserRouter>
  );
};

export default RouterFactory;
