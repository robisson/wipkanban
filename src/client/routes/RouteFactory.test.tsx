import React from "react";
import RouterFactory from "./RouterFactory";
import { createShallow } from "@material-ui/core/test-utils";
import { StaticRouter } from "react-router";
import { BrowserRouter } from "react-router-dom";
import { ShallowWrapper } from "enzyme";

describe("<RouterFactory /> Component", () => {
  let shallow: any;
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    shallow = createShallow();
  });

  it("should render a StaticRouter component when server param is defined", () => {
    wrapper = shallow(<RouterFactory server={true} location={""} />);
    expect(wrapper.type()).toEqual(StaticRouter);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render a BrowserRouter component when server param is not defined", () => {
    wrapper = shallow(<RouterFactory />);
    expect(wrapper.type()).toEqual(BrowserRouter);
    expect(wrapper).toMatchSnapshot();
  });
});
