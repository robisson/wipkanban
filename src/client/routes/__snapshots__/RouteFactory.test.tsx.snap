// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouterFactory /> Component should render a BrowserRouter component when server param is not defined 1`] = `
<BrowserRouter>
  <Routes />
</BrowserRouter>
`;

exports[`<RouterFactory /> Component should render a StaticRouter component when server param is defined 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location=""
>
  <Routes />
</StaticRouter>
`;
